exporters/otlp/otlplog/otlploggrpc/client.go

```diff
@@ -71,7 +75,18 @@ func newClient(cfg config) (*client, error) {
 
 	c.lsc = collogpb.NewLogsServiceClient(c.conn)
 
-	return c, nil
+	var err error
+	id := NextExporterID()
+	c.instrumentation, err = observ.NewInstrumentation(id, c.conn.CanonicalTarget())
+	return c, err
+}
+
+var exporterN atomic.Int64
+
+// NextExporterID returns the next unique ID for an exporter.
+func NextExporterID() int64 {
```

Comment:
This needs to be un-exported. It should not be exposed to users of this package.

```suggesion
// nextExporterID returns the next unique ID for an exporter.
func nextExporterID() int64 {
```